pkg/proxy/httputil/observability.go

```diff
@@ -86,24 +85,49 @@ func (t *wrapper) RoundTrip(req *http.Request) (*http.Response, error) {
 	attrs = append(attrs, semconv.HTTPResponseStatusCode(statusCode))
 	attrs = append(attrs, semconv.NetworkProtocolName(strings.ToLower(req.Proto)))
 	attrs = append(attrs, semconv.NetworkProtocolVersion(observability.Proto(req.Proto)))
-	attrs = append(attrs, semconv.ServerAddress(req.URL.Host))
 
 	_, port, splitErr := net.SplitHostPort(req.URL.Host)
+	var serverPort int
 	if splitErr != nil {
 		switch req.URL.Scheme {
 		case "http":
+			serverPort = 80
 			attrs = append(attrs, semconv.ServerPort(80))
 		case "https":
+			serverPort = 443
 			attrs = append(attrs, semconv.ServerPort(443))
 		}
 	} else {
 		intPort, _ := strconv.Atoi(port)
+		serverPort = intPort
 		attrs = append(attrs, semconv.ServerPort(intPort))
 	}
 
 	attrs = append(attrs, semconv.URLScheme(req.Header.Get("X-Forwarded-Proto")))
 
-	t.semConvMetricRegistry.HTTPClientRequestDuration().Record(req.Context(), end.Sub(start).Seconds(), metric.WithAttributes(attrs...))
+	// Convert method to httpconv enum
+	var methodAttr httpconv.RequestMethodAttr
+	switch req.Method {
+	case "GET":
+		methodAttr = httpconv.RequestMethodGet
+	case "POST":
+		methodAttr = httpconv.RequestMethodPost
+	case "PUT":
+		methodAttr = httpconv.RequestMethodPut
+	case "DELETE":
+		methodAttr = httpconv.RequestMethodDelete
+	case "HEAD":
+		methodAttr = httpconv.RequestMethodHead
+	case "OPTIONS":
+		methodAttr = httpconv.RequestMethodOptions
+	case "PATCH":
+		methodAttr = httpconv.RequestMethodPatch
+	default:
+		methodAttr = httpconv.RequestMethodOther
+	}
```

Comment:
This logic is redundant with the one in semconv.go. WDYT about creating a helper function that would improve maintainability?